Error.Meta cause assignment to entry in nil map

Using the JWTSecurity DSL, it seems the built in ErrJWTError has some issues when attempting to report 'missing required scope'.

Not sure where it's intended to initiate Error.MetaValues.Maybe Meta() should do it if e.MetaValues == nil?

```go
23:5:30 app         | 2016/04/12 23:05:30 [INFO] missing required scope in JWT token ctrl=version action=Show id=8RxTMJB0E1-2 missing_scope=system required_scopes=[system] scopes_in_claim=[test]
23:5:30 app         | 2016/04/12 23:05:30 [ERROR] uncaught error ctrl=version action=Show id=8RxTMJB0E1-2 id=8RxTMJB0E1-2 msg=panic: assignment to entry in nil map
github.com/goadesign/goa.(*Error).Meta(0xc820230450, 0xc820128dc8, 0x4, 0x4, 0xc820238440)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/error.go:191 +0x207
github.com/goadesign/goa/middleware/security/jwt.New.func1.1.1(0x7f9c652c8740, 0xc820230240, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/middleware/security/jwt/jwt.go:114 +0x12cc
github.com/almighty/almighty-core/app.handleSecurity.func1(0x7f9c652c8740, 0xc820230240, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/almighty/almighty-core/app/security.go:56 +0x3c9
github.com/goadesign/goa.(*Controller).MuxHandler.func1(0x7f9c652c8740, 0xc8202301e0, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/service.go:263 +0x93
github.com/goadesign/goa/middleware.Recover.func1.1(0x7f9c652c8740, 0xc8202301e0, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/middleware/recover.go:37 +0xa9
github.com/goadesign/goa/middleware.ErrorHandler.func1.1(0x7f9c652c8740, 0xc8202301e0, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/middleware/error_handler.go:18 +0xa0
github.com/goadesign/goa/middleware.LogRequest.func1.1(0x7f9c652c8740, 0xc8202301e0, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/middleware/log_request.go:63 +0xadc
github.com/goadesign/goa/middleware.RequestID.func1.1(0x7f9c652c8740, 0xc8202301b0, 0x7f9c652c9c20, 0xc8202380c0, 0xc82023a000, 0x0, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/middleware/request_id.go:50 +0x41e
github.com/goadesign/goa.(*Controller).MuxHandler.func2(0x7f9c652c9bb8, 0xc8202340b0, 0xc82023a000, 0xc8202300f0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/service.go:307 +0x915
github.com/goadesign/goa.(*mux).Handle.func1(0x7f9c652c9bb8, 0xc8202340b0, 0xc82023a000, 0x0)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/mux.go:57 +0x20e
github.com/dimfeld/httptreemux.(*TreeMux).ServeHTTP(0xc82001c2a0, 0x7f9c652c9bb8, 0xc8202340b0, 0xc82023a000)
	/home/aslak/dev/source/go/src/github.com/dimfeld/httptreemux/router.go:247 +0x86f
github.com/goadesign/goa.(*mux).ServeHTTP(0xc82012e180, 0x7f9c652c9bb8, 0xc8202340b0, 0xc82023a000)
	/home/aslak/dev/source/go/src/github.com/goadesign/goa/mux.go:83 +0x42
net/http.serverHandler.ServeHTTP(0xc82001c480, 0x7f9c652c9bb8, 0xc8202340b0, 0xc82023a000)
	/usr/lib/golang/src/net/http/server.go:1862 +0x19e
net/http.(*conn).serve(0xc820234000)
	/usr/lib/golang/src/net/http/server.go:1361 +0xbee
created by net/http.(*Server).Serve
	/usr/lib/golang/src/net/http/server.go:1910 +0x3f6
```

--- a/error.go
+++ b/error.go
@@ -182,6 +182,9 @@ func (e *Error) Error() string {
 
 // Meta adds key/value pairs to the error metadata.
 func (e *Error) Meta(keyvals ...interface{}) *Error {
+	if e.MetaValues == nil {
+		e.MetaValues = make(map[string]interface{})
+	}
 	for i := 0; i < len(keyvals); i += 2 {
 		k := keyvals[i]
 		var v interface{} = "MISSING"

--- a/error_test.go
+++ b/error_test.go
@@ -29,6 +29,20 @@ var _ = Describe("Error", func() {
 		Ω(err).ShouldNot(HaveOccurred())
 		Ω(string(b)).Should(Equal(`{"code":"invalid","status":400,"detail":"error","meta":{"what":42}}`))
 	})
+
+	Context("with no initial metadata", func() {
+		const name = "meta"
+		const val = 42
+
+		BeforeEach(func() {
+			gerr = &goa.Error{Code: code, Status: status, Detail: detail}
+		})
+
+		It("initializes the Metadata field lazily", func() {
+			Ω(func() { gerr.Meta(name, val) }).ShouldNot(Panic())
+			Ω(gerr.MetaValues).Should(HaveKeyWithValue(name, val))
+		})
+	})
 })
 
 var _ = Describe("InvalidParamTypeError", func() {

